ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/platform.py

```diff
@@ -162,23 +162,46 @@ def default_interfaces(self: SharedUtils) -> dict:
         default_interfaces set based on default_interfaces
         """
         default_interfaces = get(self.hostvars, "default_interfaces", default=[])
-
         device_platform = default(self.platform, "default")
 
+        # Re-order default_interfaces:
+        # #  platform  type
+        # 1  abc       xyz
+        # 2  abc       default
+        # 3  default   xyz
+        # 4  default   default
+        default_interfaces.sort(key=self.sort_default_interfaces)
+
         # First look for a matching default interface set that matches our platform and type
         for default_interface in default_interfaces:
```

Comment:
I feel like now that it is sorted you actually don't need both loops but only one, given that you will first get the non default platforms first and then the default ones

Reply:
Agreed.